onnxruntime/core/graph/contrib_ops/bert_defs.cc

```diff
@@ -1103,6 +1141,103 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
           GroupQueryAttentionTypeAndShapeInference(ctx, 3);
         }));
 
+constexpr const char* SparseAttention_ver1_doc = R"DOC(
+Block Sparse Attention used in Phi-3-small (https://arxiv.org/pdf/2404.14219).
+
+It is inspired by Sparse Transformers (https://arxiv.org/pdf/1904.10509) and BigBird (https://arxiv.org/pdf/2007.14062).
+
+block_mask can be used to configure sparse layout for different head.
+When number of sparse layout is 1, all heads have same sparse layout. Otherwise, different layouts are used cyclically.
```

Comment:
what is the relation between number of layout mentioned here and the interfaces (sparse_block_size and block_mask)?

Reply:
block_mask is 3D:  (num_layout, num_blocks, num_blocks).  Each head only uses one pattern (1, num_blocks, num_blocks).

When each head can have its own pattern, num_layout equals to num_heads.

For Phi-3, there are 32 heads and 8 patterns since vertial_stride=8.